File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0
        seen = set()
        i = 0
        for j in range(len(s)):
            if s[j] not in seen:
                seen.add(s[j])
                res = max(res, len(seen))
                print(seen)
            else:   
                while s[j] in seen:
                    seen.remove(s[i])
                    i += 1
                seen.add(s[j])
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-6.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let left = 0
        let res = 0
        for (let right = 0; right < prices.length; right++) {
            if (prices[right] > prices[left]) {
                res = Math.max(res, prices[right] - prices[left])
            }
            else {
                left = right
            }
        }
        return res
    }
}
